device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -316,6 +316,15 @@ else if (message != null)
                         {
                             clientException = e.toIotHubClientException();
                         }
+                        else
```

Comment:
Do you think it's necessary to add a comment here describing which scenario for instance can be caught by this block?

Reply:
Good idea, sure